homeassistant/components/mqtt/device_tracker.py

```diff
@@ -137,8 +137,7 @@ def message_received(msg: ReceiveMessage) -> None:
             elif payload == self._config[CONF_PAYLOAD_RESET]:
                 self._location_name = None
             else:
-                assert isinstance(msg.payload, str)
```

Comment:
Would it make sense to guard this with if TYPE_CHECKING: instead?

Reply:
That could be an alternative, may be that is a better plan.

Reply:
Drafting the PR because of the above.